### PR TITLE
Fix agent charm on focal

### PR DESCRIPTION
**Pull request by plars**
## Description
Thanks to @nancyc12 for alerting me to this last night. There were some changes that required rebuilding the agent charm recently, which is really rare. These weren't set up to autobuild because they are almost never needed, so I've added that, and requested ownership change of these charms on forums.charmhub.io. I've pushed an update for now that got her unblocked for the system she was trying to fix, but there's another issue that will happen if the agent host is on focal. Most of our agent hosts are on jammy or above now, but one of them is still on focal. This should fix that.

## Resolved issues
No issue filed that I'm aware of

## Documentation
N/A

## Web service API changes
N/A

## Tests
I've tried this update to pip in a focal container and it does what is needed, but I intend to do an end-to-end test with a built version of the charm before deploying it widely.

### CODE REVIEW

File: .github/workflows/agent-charm-release-edge.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Release testflinger-agent charm to latest/edge
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - agent/testflinger-agent-charm/**
```
Comment (nancyc12) [inline .github/workflows/agent-charm-release-edge.yml:8]: I'm wondering if the path should be like this. Please correct me if I misunderstood.
```suggestion
      - agent/charms/testflinger-agent-charm/**
```

File: .github/workflows/agent-charm-release-edge.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@2.6.0
+        with:
+          charm-path: server/charm
```
Comment (nancyc12) [inline .github/workflows/agent-charm-release-edge.yml:25]: `charm-path` also confuses me.
```suggestion
          charm-path: agent/charms/testflinger-agent-charm
```

File: .github/workflows/agent-host-charm-release-edge.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+name: Release testflinger-agent charm to latest/edge
```
Comment (nancyc12) [inline .github/workflows/agent-host-charm-release-edge.yml:1]:
```suggestion
name: Release testflinger-agent-host charm to latest/edge
```

File: .github/workflows/agent-host-charm-release-edge.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - agent/testflinger-agent-charm/**
```
Comment (nancyc12) [inline .github/workflows/agent-host-charm-release-edge.yml:8]:
```suggestion
      - agent/charms/testflinger-agent-host-charm/**
```

File: .github/workflows/agent-host-charm-release-edge.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@2.6.0
+        with:
+          charm-path: server/charm
```
Comment (nancyc12) [inline .github/workflows/agent-host-charm-release-edge.yml:25]:
```suggestion
          charm-path: agent/charms/testflinger-agent-host-charm
```

**Review by nancyc12 — CHANGES_REQUESTED** [review]
Thank you for making this auto-release workflow!
I don't have much experience with GitHub workflow but a few lines're confusing me.

**Comment by plars** [comment]
@nancyc12 I forgot this had not landed yet, thanks for finding my dumb copy/paste errors :) Can you recheck it when you get a moment?

File: .github/workflows/agent-charm-release-edge.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+name: Release testflinger-agent charm to latest/edge
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - agent/charms/testflinger-agent-charm/**
```
Comment (nadzyah) [inline .github/workflows/agent-charm-release-edge.yml:8]: I suggest to run this action also when there are updates to this CI file

File: .github/workflows/agent-host-charm-release-edge.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+name: Release testflinger-agent-host charm to latest/edge
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - agent/charms/testflinger-agent-host-charm/**
```
Comment (nadzyah) [inline .github/workflows/agent-host-charm-release-edge.yml:8]: Same here

File: .github/workflows/agent-charm-release-edge.yml (hunk 6 above)
Comment (plars) [reply .github/workflows/agent-charm-release-edge.yml:8]: I was going to say we could just use workflow dispatch if we really need to trigger a build, but I can see some value in also using this as a test of any change that might get made to the workflow file in the future. Good idea!
Fixed

File: .github/workflows/agent-host-charm-release-edge.yml (hunk 7 above)
Comment (plars) [reply .github/workflows/agent-host-charm-release-edge.yml:8]: done

Comment (nancyc12) [reply .github/workflows/agent-host-charm-release-edge.yml:8]: same question here

File: .github/workflows/agent-charm-release-edge.yml (hunk 6 above)
Comment (nancyc12) [reply .github/workflows/agent-charm-release-edge.yml:8]: @plars did you implement it elsewhere or just forgot to push it here?
```suggestion
    paths:
      - agent/charms/testflinger-agent-charm/**
      - .github/workflows/agent-charm-release-edge.yml
```

File: .github/workflows/agent-host-charm-release-edge.yml (hunk 7 above)
Comment (nancyc12) [reply .github/workflows/agent-host-charm-release-edge.yml:8]: same question here

File: .github/workflows/agent-charm-release-edge.yml (hunk 6 above)
Comment (plars) [reply .github/workflows/agent-charm-release-edge.yml:8]: @nancyc12 @nadzyah I could swear I pushed it... guess not it was sitting in my local branch feeling like I forgot about it :)

**Review by nancyc12 — APPROVED** [review]
Thank you!

**Pull request merged**